syntax = "proto3";

package testing.proto;

service Echo {
    rpc UnaryEchoTest (EchoRequest) returns (EchoResponse);
    rpc ServerStreamEchoTest (EchoRequest) returns (stream EchoResponse);
}

message EchoRequest {
    string message = 1;
    int32 expected_responses = 2;
}

message EchoResponse {
    string message = 1;
    int32 response_number = 2;
}
